spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/GcpCloudSqlProperties.java

```diff
@@ -127,4 +129,12 @@ public Path getCredentialsLocation() {
 	public void setCredentialsLocation(Path credentialsLocation) {
 		this.credentialsLocation = credentialsLocation;
 	}
+
+	public DatabaseType getDatabaseType() {
```

Comment:
`@NotNull` ?

Reply:
I think you mean in the setter. Agreed. We should probably also add assertions to all the setters to prevent setting to `null`.

Reply:
Well, if `jdbcDriver` is an alternative, then this can be `null`.
So, never mind.
Thanks

Although I'm not sure which other driver end-user can choose, if we have only MySQL and PostgreSQL choice...

Reply:
One example I like to think of is MySQL 8: the driver is now `com.mysql.cj.jdbc.Driver`. If users add that dependency on their side and set the driver class name to that, it should work with that MySQL version too.
For App Engine Standard, we also found that we had to select a different driver than the default MySQL one (`com.mysql.jdbc.GoogleDriver`).
There might be other edge cases in other environments there that we don't know about. So, it's nice to give users the flexibility to change this if they need to.

`DatabaseType` shouldn't be set to `null`. If the user provides the JDBC URL and driver class name,  we don't need to check its value, but in any other case we do. `databaseType` already has a default value, there's no good reason for a user to be setting `null` I can think of.

Reply:
Initially I was thinking of adding null checks to every setter, but since we have plenty, I started wondering whether it was worth it and then decided not to.